### PR TITLE
Feature#20 add styles to feature reports view

### LINKED ISSUE #20: Add styles to feature Reports View
# REQUIREMENTS
In this issue all styles are going to be added in the Reports View

# DESIGN
Provide references to images of the design, how it looks like now, and how it should look like.

# Acceptance Criteria
List all required acceptance criteria
- [x] The final result should be similar to the design

### PR DESCRIPTION
**Pull request by mariormt17**
## Description: Why did you change/implement that?

In this PR we added the styles in the views of the attendance reports per day, average time employees checked in/out by month and absence report by month.

## Testing: Help me how can I test or look at the changes?

imagen

### CODE REVIEW

File: reloj-checador/app/controllers/reports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    #   .paginate(page: params[:page], per_page: 1)
+    
+    @abscenes = Attendace.where("#{(!date.nil?) ? "attendaces.date = '#{date}'" : ""}")
+                         .select("attendaces.private_number, date, E.company_id, COUNT( CASE WHEN check_type = 'IN' THEN 1 END)")
+                         .group("attendaces.private_number, date, E.company_id")
+                         .joins("JOIN employees E ON E.private_number = attendaces.private_number")
+                         .paginate(page: params[:page], per_page: 1)
+    p "@abscenes.to_json #{@abscenes.to_json}"
+  end
+end
```
Comment (YilKanda) [inline reloj-checador/app/controllers/reports_controller.rb]: Add a blank line.

File: reloj-checador/app/views/employees/index.html.erb [hunk 2]
```diff
@@ -24,5 +66,6 @@
         </tbody>
       </table>
     </div>
+>>>>>>> 2180a79516cefc963819e448676c43d41cf92aa3
   </div>
 </div>
```
Comment (YilKanda) [inline reloj-checador/app/views/employees/index.html.erb]: Add a blank line.

File: reloj-checador/app/views/reports/avg_time_by_employees.xlsx.axlsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      "#{avg["yyyy"].to_s.split(".0")[0]}", #B
+      "#{avg["time_in"]}", #C
+      "#{avg["time_out"]}"  #D
+    ], style: [wrap_text,   #A
+               wrap_text,   #B
+               wrap_text,   #C
+               wrap_text    #D
+              ]
+  end
+end
```
Comment (YilKanda) [inline reloj-checador/app/views/reports/avg_time_by_employees.xlsx.axlsx]: Add a blank line.

File: reloj-checador/app/views/employees/index.html.erb [hunk 4]
```diff
@@ -3,6 +3,48 @@
...
+              <% end %>
+               <%= link_to employee, method: :delete, data: { confirm: 'Are you sure?' }, :class => "btn btn-danger" do %>
+                <i class="fas fa-trash"></i>Destroy
+              <% end %>
+            </td>
+          </tr>
+        <% end %>
+      </tbody>
+    </table>
+=======
```
Comment (YilKanda) [inline reloj-checador/app/views/employees/index.html.erb]: I notice that here a conflict occurred between the branches and the master and the lines indicating the conflicting lines were not removed.

```
<<<<<<< HEAD

 =======

>>>>>>> 2180a79516cefc963819e448676c43d41cf92aa3
```

File: reloj-checador/app/views/reports/att_by_day.xlsx.axlsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  # Create entries for each item
+  @attendaces.order(date: :asc).each do |attendace|
+    sheet.add_row [
+      "#{attendace&.date.strftime('%Y-%m-%d')}", #A
+      attendace&.date_count #B
+    ], style: [wrap_text,   #A
+               wrap_text   #B
+              ]
+  end
+end
```
Comment (YilKanda) [inline reloj-checador/app/views/reports/att_by_day.xlsx.axlsx]: Add a blank line.

**Review by YilKanda — CHANGES_REQUESTED** [review]
Apparently you had conflicts in git, and did not delete the conflict lines.

File: reloj-checador/app/controllers/reports_controller.rb (hunk 1 above)
Comment (mariormt17) [reply reloj-checador/app/controllers/reports_controller.rb]: Thanks

File: reloj-checador/app/views/employees/index.html.erb (hunk 2 above)
Comment (mariormt17) [reply reloj-checador/app/views/employees/index.html.erb]: Thanks

File: reloj-checador/app/views/reports/avg_time_by_employees.xlsx.axlsx (hunk 3 above)
Comment (mariormt17) [reply reloj-checador/app/views/reports/avg_time_by_employees.xlsx.axlsx]: Thanks

File: reloj-checador/app/views/reports/att_by_day.xlsx.axlsx (hunk 5 above)
Comment (mariormt17) [reply reloj-checador/app/views/reports/att_by_day.xlsx.axlsx]: Thanks

File: reloj-checador/app/views/employees/index.html.erb (hunk 4 above)
Comment (mariormt17) [reply reloj-checador/app/views/employees/index.html.erb]: Omg! it slipped my mind jajaja thanks

**Review by EduardoGHdez — COMMENTED** [review]
Hey 👋🏽 

Just noticed you have some commits in your PR that do not seem to be part of your changes. Like the merge-commits, for example

Do you mind removing them from your branch and rebasing with latest master? 😄

File: reloj-checador/Gemfile [hunk 6]
```diff
@@ -32,6 +32,13 @@ gem 'bootsnap', '>= 1.4.4', require: false
 gem 'rspec-rails'
 gem 'devise'
 
+#generate xls
+gem 'axlsx', '= 2.0.1'
+gem 'axlsx_rails'
+
+gem 'will_paginate', '~> 3.1.0'
+gem 'will_paginate-bootstrap4'
+
```
Comment (EduardoGHdez) [inline reloj-checador/Gemfile]: When adding a new dependency, worth committing in isolation and explain in the commit message why are those dependencies needed or added 😄 

That would help other teammates to understand why are we adding them and not only teammates but future contributors.

I know this is just a kata/excersice. But consider this for future contributions 😄

File: reloj-checador/spec/requests/reports_request_spec.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  describe "GET /absence_by_month" do
+    it "returns http success" do
+      get "/reports/absence_by_month"
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+end
```
Comment (EduardoGHdez) [inline reloj-checador/spec/requests/reports_request_spec.rb:26]: ⭐

File: reloj-checador/Gemfile (hunk 6 above)
Comment (mariormt17) [reply reloj-checador/Gemfile]: Thank you so much, we will consider that in the future😊

**Pull request merged**